### ISSUE TITLE
Can I change task_id in flight?

### ISSUE THREAD

**Issue by ankerstal** [logstash-plugins/logstash-filter-aggregate#83]
This is maybe not the correct place to ask this but I found it very hard to get examples on how to use aggregate. I have a case where I want to follow a event after the unique identifier changes. Look at these logs:
```
Feb 21 11:01:31 m20 sshd[11097]: Connection from 172.18.37.40 port 38944 on 172.18.5.20 port 22
[...]
Feb 21 11:01:32 m20 sshd[11097]: User child is on pid 11104
Feb 21 11:01:32 m20 sshd[11104]: Starting session: shell on pts/1 for pan from 172.18.37.40 port 38944 id 0
```
I want to aggregate on the pid, but as you can see the last log line have a new pid. Is it possible to do this?

Thanks!

**Comment by fbaligand** [comment]
Hi @ankerstal 

First, I'm very surprised you found hard to get examples. Aggregate is the plugin which have the greatest number of examples (5) in official documentation : 
https://www.elastic.co/guide/en/logstash/current/plugins-filters-aggregate.html

Then, I think you could reach this goal by stopping the first aggregate map, then starting a new aggregate map based on new id.

Here's a configuration that should meet your needs : 
```
if [message] =~ "User child is on pid" {
  aggregate {
    task_id => "%{pid}"
    end_of_task => true
    code => "event.set('field1', map['field1']) ; ..."
  }
  aggregate {
    task_id => "%{child_pid}"
    code => "map['field1'] = event.get('field1'); ..."
  }
}

```

**Comment by ankerstal** [comment]
Thank you @fbaligand! 

Yes, the documentation is fine, so is the examples. I have stared at them for several days :D 

I thought that task_id was the unique identifier that would link events into my aggregate. So what I dont understand is how to continue to follow a event if when the pid changes.

Thanks again for your time!

**Comment by fbaligand** [comment]
First, you thought right : task_id is the unique identifier that link events into an aggregate map.

In the example configuration I gave to you, I use the particular log line where you have both "pid" and "child_pid", to end aggregate map based on "pid" and start a new aggregate map based on "child_pid".

In the first aggregate filter, I end "pid" aggregation, and I copy all aggregate map fields to current event.
In the second aggregate filter, I start a new "child_pid" aggregation, and I copy all current event fields to aggregate map.

So basically, I don't change task_id, but I end one and start another. Because, it's not possible to literally change current aggregation map "task_id". But in the end, it lets the same thing.

**Comment by ankerstal** [comment]
Ah, now I get it. Thanks you!

**Comment by fbaligand** [comment]
Nice !

**Comment by fbaligand** [comment]
As the issue is resolved, I close the issue.
